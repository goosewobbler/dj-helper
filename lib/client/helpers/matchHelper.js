"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fp_1 = require("lodash/fp");
var normalise = function (value) { return fp_1.deburr(value.toLowerCase()); };
var getNeedleSubstring = function (needle, haystack) {
    var normalisedNeedle = normalise(needle);
    var normalisedHaystack = normalise(haystack);
    var indexOfFirstNeedleCharacter = normalisedHaystack.indexOf(normalisedNeedle[0]);
    var length = 0;
    if (indexOfFirstNeedleCharacter > -1) {
        length = 1;
        while (length < normalisedNeedle.length &&
            normalisedHaystack[indexOfFirstNeedleCharacter + length] === normalisedNeedle[length]) {
            length += 1;
        }
    }
    return { indexOfFirstNeedleCharacter: indexOfFirstNeedleCharacter, length: length };
};
var getMatchesRecursive = function (needle, haystack) {
    var _a = getNeedleSubstring(needle, haystack), indexOfFirstNeedleCharacter = _a.indexOfFirstNeedleCharacter, length = _a.length;
    if (length === 0) {
        return [haystack];
    }
    var start = haystack.substr(0, indexOfFirstNeedleCharacter);
    var matched = haystack.substr(indexOfFirstNeedleCharacter, length);
    var end = haystack.substr(indexOfFirstNeedleCharacter + length);
    return [start, { matched: matched }].concat(getMatchesRecursive(needle.substr(length), end)).filter(Boolean);
};
var getMatches = function (needle, haystack) {
    var trimmedNeedle = fp_1.trim(needle);
    var matches = getMatchesRecursive(trimmedNeedle, haystack);
    var matchedParts = matches.filter(function (match) { return typeof match === 'object'; });
    var combinedMatchedParts = matchedParts.reduce(function (acc, match) { return acc + match.matched; }, '');
    if (combinedMatchedParts.length !== trimmedNeedle.length) {
        return [haystack];
    }
    return matches;
};
var getMatchesWithAlternatives = function (needle, haystack, alternatives) {
    var matches = getMatches(needle, haystack);
    if (fp_1.some(fp_1.isObject)(matches)) {
        return matches;
    }
    var alternativeMatches = alternatives.map(function (alternative) { return getMatches(needle, alternative); });
    return fp_1.find(fp_1.some(fp_1.isObject), alternativeMatches);
};
exports.default = getMatchesWithAlternatives;
//# sourceMappingURL=matchHelper.js.map