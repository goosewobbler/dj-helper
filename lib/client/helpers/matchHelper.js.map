{"version":3,"file":"matchHelper.js","sourceRoot":"","sources":["../../../src/client/helpers/matchHelper.ts"],"names":[],"mappings":";;AAAA,gCAA+D;AAE/D,IAAM,SAAS,GAAG,UAAC,KAAa,IAAa,OAAA,WAAM,CAAC,KAAK,CAAC,WAAW,EAAE,CAAC,EAA3B,CAA2B,CAAC;AAEzE,IAAM,kBAAkB,GAAG,UAAC,MAAc,EAAE,QAAgB;IAC1D,IAAM,gBAAgB,GAAG,SAAS,CAAC,MAAM,CAAC,CAAC;IAC3C,IAAM,kBAAkB,GAAG,SAAS,CAAC,QAAQ,CAAC,CAAC;IAC/C,IAAM,2BAA2B,GAAG,kBAAkB,CAAC,OAAO,CAAC,gBAAgB,CAAC,CAAC,CAAC,CAAC,CAAC;IAEpF,IAAI,MAAM,GAAG,CAAC,CAAC;IAEf,IAAI,2BAA2B,GAAG,CAAC,CAAC,EAAE;QACpC,MAAM,GAAG,CAAC,CAAC;QACX,OACE,MAAM,GAAG,gBAAgB,CAAC,MAAM;YAChC,kBAAkB,CAAC,2BAA2B,GAAG,MAAM,CAAC,KAAK,gBAAgB,CAAC,MAAM,CAAC,EACrF;YACA,MAAM,IAAI,CAAC,CAAC;SACb;KACF;IAED,OAAO,EAAE,2BAA2B,6BAAA,EAAE,MAAM,QAAA,EAAE,CAAC;AACjD,CAAC,CAAC;AAEF,IAAM,mBAAmB,GAAG,UAAC,MAAc,EAAE,QAAgB;IACrD,IAAA,yCAA8E,EAA5E,4DAA2B,EAAE,kBAA+C,CAAC;IAErF,IAAI,MAAM,KAAK,CAAC,EAAE;QAChB,OAAO,CAAC,QAAQ,CAAC,CAAC;KACnB;IAED,IAAM,KAAK,GAAG,QAAQ,CAAC,MAAM,CAAC,CAAC,EAAE,2BAA2B,CAAC,CAAC;IAC9D,IAAM,OAAO,GAAG,QAAQ,CAAC,MAAM,CAAC,2BAA2B,EAAE,MAAM,CAAC,CAAC;IACrE,IAAM,GAAG,GAAG,QAAQ,CAAC,MAAM,CAAC,2BAA2B,GAAG,MAAM,CAAC,CAAC;IAElE,OAAO,CAAC,KAAK,EAAE,EAAE,OAAO,SAAA,EAAE,CAAC,CAAC,MAAM,CAAC,mBAAmB,CAAC,MAAM,CAAC,MAAM,CAAC,MAAM,CAAC,EAAE,GAAG,CAAC,CAAC,CAAC,MAAM,CAAC,OAAO,CAAC,CAAC;AACtG,CAAC,CAAC;AAEF,IAAM,UAAU,GAAG,UAAC,MAAc,EAAE,QAAgB;IAClD,IAAM,aAAa,GAAG,SAAI,CAAC,MAAM,CAAC,CAAC;IACnC,IAAM,OAAO,GAAG,mBAAmB,CAAC,aAAa,EAAE,QAAQ,CAAC,CAAC;IAC7D,IAAM,YAAY,GAAG,OAAO,CAAC,MAAM,CAAC,UAAA,KAAK,IAAI,OAAA,OAAO,KAAK,KAAK,QAAQ,EAAzB,CAAyB,CAA+B,CAAC;IACtG,IAAM,oBAAoB,GAAG,YAAY,CAAC,MAAM,CAAC,UAAC,GAAG,EAAE,KAAK,IAAK,OAAA,GAAG,GAAG,KAAK,CAAC,OAAO,EAAnB,CAAmB,EAAE,EAAE,CAAC,CAAC;IAE1F,IAAI,oBAAoB,CAAC,MAAM,KAAK,aAAa,CAAC,MAAM,EAAE;QACxD,OAAO,CAAC,QAAQ,CAAC,CAAC;KACnB;IAED,OAAO,OAAO,CAAC;AACjB,CAAC,CAAC;AAEF,IAAM,0BAA0B,GAAG,UAAC,MAAc,EAAE,QAAgB,EAAE,YAAsB;IAC1F,IAAM,OAAO,GAAG,UAAU,CAAC,MAAM,EAAE,QAAQ,CAAC,CAAC;IAE7C,IAAI,SAAI,CAAC,aAAQ,CAAC,CAAC,OAAO,CAAC,EAAE;QAC3B,OAAO,OAAO,CAAC;KAChB;IAED,IAAM,kBAAkB,GAAG,YAAY,CAAC,GAAG,CAAC,UAAA,WAAW,IAAI,OAAA,UAAU,CAAC,MAAM,EAAE,WAAW,CAAC,EAA/B,CAA+B,CAAC,CAAC;IAC5F,OAAO,SAAI,CAAC,SAAI,CAAC,aAAQ,CAAC,EAAE,kBAAkB,CAAC,CAAC;AAClD,CAAC,CAAC;AAEF,kBAAe,0BAA0B,CAAC"}