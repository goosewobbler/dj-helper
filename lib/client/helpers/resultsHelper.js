"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fp_1 = require("lodash/fp");
var React = require("react");
var matchHelper_1 = require("./matchHelper");
var numberOfMatchedRegions = function (item) {
    return item.matches.reduce(function (acc, match) { return acc + (fp_1.isObject(match) ? 1 : 0); }, 0);
};
var itemStartsWithSignificantMatch = function (item) {
    return !(fp_1.isObject(item.matches[0]) && item.matches[0].matched.length > 1);
};
var alphabeticalSort = fp_1.sortBy(fp_1.get('displayName'));
var numberOfRegionsSort = fp_1.sortBy(numberOfMatchedRegions);
var startsWithSignificantMatchSort = fp_1.sortBy(itemStartsWithSignificantMatch);
var sortItems = function (list) { return startsWithSignificantMatchSort(numberOfRegionsSort(alphabeticalSort(list))); };
var createItemWithMatches = function (item, searchValue) {
    return fp_1.assign(item, {
        matches: matchHelper_1.default(searchValue, item.displayName, item.alternatives),
    });
};
var itemMatchesSearchValue = function (item) { return fp_1.some(fp_1.isObject)(item.matches); };
var findOrderedSearchResults = function (items, searchValue) {
    var itemsWithMatches = items.map(function (item) { return createItemWithMatches(item, searchValue); });
    var results = sortItems(itemsWithMatches.filter(itemMatchesSearchValue));
    return results.map(function (item) {
        return fp_1.assign(item, {
            highlighted: item.matches.map(function (match, index) {
                if (match.matched) {
                    return React.createElement("mark", { key: "match-" + index }, match.matched.replace(/ /g, '-'));
                }
                return match.replace(/ /g, '-');
            }),
        });
    });
};
exports.default = findOrderedSearchResults;
//# sourceMappingURL=resultsHelper.js.map