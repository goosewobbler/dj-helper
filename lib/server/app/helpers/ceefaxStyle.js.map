{"version":3,"file":"ceefaxStyle.js","sourceRoot":"","sources":["../../../../src/server/app/helpers/ceefaxStyle.ts"],"names":[],"mappings":";;AAAA,IAAM,WAAW,GAAG,6hBAqBX,CAAC;AAEV,kBAAe,WAAW,CAAC"}