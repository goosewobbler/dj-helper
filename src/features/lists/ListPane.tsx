import React, { ReactElement } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { List } from './List';
import {
  createList,
  editList,
  revertEditList,
  finishEditList,
  deleteList,
  selectLists,
  updateListTitle,
} from './listsSlice';

export function ListPane(): ReactElement {
  const lists = useSelector(selectLists);
  const dispatch = useDispatch();

  return (
    <div>
      <ol className="listPane">
        {lists.map(
          (list): ReactElement => (
            <List
              key={list.id}
              id={list.id}
              title={list.title}
              editing={list.editing}
              onTitleChange={(id: number, title: string) => dispatch(updateListTitle({ id, title }))}
              onClickEdit={(id: number) => dispatch(editList({ id }))}
              onClickDelete={(id: number) => dispatch(deleteList({ id }))}
              onEditingComplete={() => dispatch(finishEditList())}
              onEditingCancelled={() => dispatch(revertEditList())}
            />
          ),
        )}
      </ol>
      <div className="">
        <button
          className="p-1 bg-blue-100 border"
          type="button"
          aria-label="New List"
          onClick={() => {
            dispatch(revertEditList());
            dispatch(createList());
          }}
        >
          New List
        </button>
      </div>
    </div>
  );
}
