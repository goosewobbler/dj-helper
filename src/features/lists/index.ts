import type { AppState, AppStore, List, Track } from '../../common/types.js';

const initialListTitle = 'New List';

function reorderTracks(tracks: List['tracks'], trackIdToMove: Track['id'], newIndex: number) {
  const indexOfTrackIdToMove = tracks.indexOf(trackIdToMove);
  if (indexOfTrackIdToMove === -1) {
    return tracks;
  }

  // clone array and remove track
  const otherTracks = Array.from(tracks);
  otherTracks.splice(indexOfTrackIdToMove, 1);

  // insert track at the desired index
  const tracksBefore = otherTracks.slice(0, newIndex);
  const tracksAfter = otherTracks.slice(newIndex);
  return tracksBefore.concat([trackIdToMove]).concat(tracksAfter);
}

export const initialState: List[] = [
  {
    id: 0,
    title: 'Look a set list',
    tracks: [],
    editing: false,
    active: true,
  },
  {
    id: 1,
    title: 'Wow such list',
    tracks: [],
    editing: false,
    active: false,
  },
];

export const handlers = (store: AppStore) => ({
  'LIST:CREATE': () => {
    store.setState((state) => {
      const newList: List = {
        id: state.lists.length,
        title: initialListTitle,
        tracks: [],
        editing: true,
        active: true,
      };
      return { lists: [...state.lists.map((list) => ({ ...list, active: false })), newList] };
    });
  },
  'LIST:UPDATE_TITLE': ({ id, title }: { id: number; title: string }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) => (list.id === id ? { ...list, title } : list)),
    })),
  'LIST:DELETE': ({ id }: { id: number }) =>
    store.setState((state) => ({
      lists: state.lists.filter((list) => list.id !== id).map((list, index) => ({ ...list, id: index })),
    })),
  'LIST:EDIT': ({ id }: { id: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) => (list.id === id ? { ...list, editing: true, oldTitle: list.title } : list)),
    })),
  'LIST:EDIT_COMPLETE': () =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.editing === true ? { ...list, editing: undefined, oldTitle: undefined } : list,
      ),
    })),
  'LIST:EDIT_REVERT': () =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.editing === true
          ? { ...list, editing: undefined, title: list.oldTitle ?? initialListTitle, oldTitle: undefined }
          : list,
      ),
    })),
  'LIST:TOGGLE_ACTIVE': ({ id }: { id: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.id === id ? { ...list, active: !list.active } : { ...list, active: false },
      ),
    })),
  'LIST:ADD_TRACK': ({ trackId }: { trackId: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) => (list.active ? { ...list, tracks: [...list.tracks, trackId] } : list)),
    })),
  'LIST:REMOVE_TRACK': ({ trackId }: { trackId: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.active ? { ...list, tracks: list.tracks.filter((track) => track !== trackId) } : list,
      ),
    })),
  'LIST:MOVE_TRACK': ({ trackId, newIndex }: { trackId: number; newIndex: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.active ? { ...list, tracks: reorderTracks(list.tracks, trackId, newIndex) } : list,
      ),
    })),
});

export const selectLists = (state: Partial<AppState>) => state.lists as List[];

export const selectActiveList = (state: Partial<AppState>): List => state.lists?.find((list) => list.active) as List;

export const selectListById =
  (id: number) =>
  (state: Partial<AppState>): List =>
    state.lists?.find((list) => list.id === id) as List;

export const trackIsOnActiveList =
  ({ trackId }: { trackId: number }) =>
  (state: Partial<AppState>): boolean =>
    (state.lists?.find((list) => list.active) as List)?.tracks.some((track) => track === trackId);

export const getNextTrackOnList =
  ({ id, currentTrackId }: { id: number; currentTrackId: number }) =>
  (state: Partial<AppState>): Track['id'] | undefined => {
    const list = state.lists?.find((stateList) => stateList.id === id) as List;
    const listTracks = (state.lists?.find((stateList) => stateList.id === id) as List).tracks;
    if (!list) {
      return undefined;
    }
    const currentTrackIndex = list.tracks.findIndex((track) => track === currentTrackId);
    return listTracks[currentTrackIndex + 1];
  };

export const getPreviousTrackOnList =
  ({ id, currentTrackId }: { id: number; currentTrackId: number }) =>
  (state: Partial<AppState>): Track['id'] | undefined => {
    const list = state.lists?.find((stateList) => stateList.id === id) as List;
    const listTracks = (state.lists?.find((stateList) => stateList.id === id) as List).tracks;
    if (!list) {
      return undefined;
    }
    const currentTrackIndex = listTracks.findIndex((track) => track === currentTrackId);
    return listTracks[currentTrackIndex - 1];
  };
