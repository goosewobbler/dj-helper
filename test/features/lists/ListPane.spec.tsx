import * as React from 'react';
import { fireEvent, render, RenderResult } from '../../helpers/integration';
import { ListPane } from '../../../src/features/lists/ListPane';

describe('ListPane', () => {
  let listPane: RenderResult;

  describe('initial render', () => {
    beforeEach(() => {
      listPane = render(<ListPane />);
    });

    it('should render no lists', () => {
      expect(listPane.queryByRole('listitem')).not.toBeInTheDocument();
    });

    it('should render a new list button', () => {
      expect(listPane.queryByRole('button', { name: /New List/i })).toBeInTheDocument();
    });

    it('should render the expected html', () => {
      expect(listPane.container).toMatchSnapshot();
    });

    describe('when the new list button is clicked', () => {
      beforeEach(() => {
        const newListBtn = listPane.getByRole('button', { name: /New List/i });
        fireEvent.click(newListBtn);
      });

      it('should render a new list', () => {
        expect(listPane.queryAllByRole('listitem').length).toEqual(1);
        expect(listPane.queryAllByTestId('title').map((title) => title.textContent)).toEqual(['List Title']);
      });

      it('should render the expected html', () => {
        expect(listPane.container).toMatchSnapshot();
      });
    });
  });

  describe('given a number of lists', () => {
    beforeEach(() => {
      listPane = render(<ListPane />, {
        preloadedState: {
          lists: [
            { id: 1, title: 'test list one', tracks: [] },
            { id: 2, title: 'test list two', tracks: [] },
            { id: 3, title: 'test list three', tracks: [] },
          ],
          tracks: [],
          browsers: [],
        },
      });
    });

    it('should render the expected lists', () => {
      expect(listPane.queryAllByRole('listitem').length).toEqual(3);
      expect(listPane.queryAllByTestId('title').map((title) => title.textContent)).toEqual([
        'test list one',
        'test list two',
        'test list three',
      ]);
    });

    it('should render a new list button', () => {
      expect(listPane.queryByRole('button', { name: /New List/i })).toBeInTheDocument();
    });

    it('should render the expected html', () => {
      expect(listPane.container).toMatchSnapshot();
    });

    describe('when the new list button is clicked', () => {
      beforeEach(() => {
        const newListBtn = listPane.getByRole('button', { name: /New List/i });
        fireEvent.click(newListBtn);
      });

      it('should render a new list', () => {
        expect(listPane.queryAllByRole('listitem').length).toEqual(4);
        expect(listPane.queryAllByTestId('title').map((title) => title.textContent)).toEqual([
          'test list one',
          'test list two',
          'test list three',
          'List Title',
        ]);
      });

      it('should render the expected html', () => {
        expect(listPane.container).toMatchSnapshot();
      });
    });

    describe('when a list delete button is clicked', () => {
      beforeEach(() => {
        const listDeleteBtns = listPane.getAllByRole('button', { name: /Delete/i });
        fireEvent.click(listDeleteBtns[1]);
      });

      it('should render the expected lists', () => {
        expect(listPane.queryAllByRole('listitem').length).toEqual(2);
        expect(listPane.queryAllByTestId('title').map((title) => title.textContent)).toEqual([
          'test list one',
          'test list three',
        ]);
      });

      it('should render the expected html', () => {
        expect(listPane.container).toMatchSnapshot();
      });
    });

    describe('when a list edit button is clicked', () => {
      beforeEach(() => {
        const listEditBtns = listPane.getAllByRole('button', { name: /Edit/i });
        fireEvent.click(listEditBtns[1]);
      });

      it('should render a text input with the expected value', () => {
        expect(listPane.getByLabelText('List Title').getAttribute('value')).toEqual('test list two');
      });

      it('should render the expected html', () => {
        expect(listPane.container).toMatchSnapshot();
      });

      describe('and the edit is cancelled', () => {
        beforeEach(() => {
          const input = listPane.getByLabelText('List Title');
          fireEvent.change(input, { target: { value: 'new title for test list two' } });
          fireEvent.keyDown(input, { key: 'Escape', keyCode: 27 });
        });

        it('should render a List with the expected title', () => {
          expect(listPane.getAllByTestId('title')[1].textContent).toEqual('test list two');
        });

        it('should render the expected html', () => {
          expect(listPane.container).toMatchSnapshot();
        });
      });

      describe('and the edit is completed', () => {
        beforeEach(() => {
          const input = listPane.getByLabelText('List Title');
          fireEvent.change(input, { target: { value: 'new title for test list two' } });
          fireEvent.keyDown(input, { key: 'Enter', keyCode: 13 });
        });

        it('should render a List with the expected title', () => {
          expect(listPane.getAllByTestId('title')[1].textContent).toEqual('new title for test list two');
        });

        it('should render the expected html', () => {
          expect(listPane.container).toMatchSnapshot();
        });
      });

      describe('and an invalid title stops the edit from being completed', () => {
        beforeEach(() => {
          const input = listPane.getByLabelText('List Title');
          fireEvent.change(input, { target: { value: '' } });
          fireEvent.keyDown(input, { key: 'Enter', keyCode: 13 });
        });

        it('should render a text input with the expected value', () => {
          expect(listPane.getByLabelText('List Title').getAttribute('value')).toEqual('');
        });

        it('should render the expected html', () => {
          expect(listPane.container).toMatchSnapshot();
        });
      });
    });
  });
});

// describe('when the edit button is clicked', () => {

//     describe('and an invalid title is entered', () => {
//       beforeEach(() => {
//         const input = list.getByLabelText('List Title');
//         fireEvent.change(input, { target: { value: '' } });
//         fireEvent.keyDown(input, { key: 'Enter', keyCode: 13 });
//       });

//       it('should render a text input with the expected value', () => {
//         expect(list.getByLabelText('List Title').getAttribute('value')).toEqual('');
//       });

//       it('should render the expected html', () => {
//         expect(list.container).toMatchSnapshot();
//       });
//     });
