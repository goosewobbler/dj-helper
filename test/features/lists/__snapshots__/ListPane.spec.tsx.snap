// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPane given a number of lists should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list one
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-2"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list two
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-3"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list three
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane given a number of lists when a list delete button is clicked should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list one
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-2"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list three
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane given a number of lists when a list edit button is clicked and an invalid title stops the edit from being completed should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list one
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-2"
      >
        <span
          class="title"
          data-testid="title"
        >
          <label
            for="list-title"
          >
            List Title
            <input
              class="ring-4 ring-opacity-30 ring-red-300"
              id="list-title"
              placeholder="List Title"
              type="text"
              value=""
            />
          </label>
        </span>
      </li>
      <li
        class="list"
        id="list-3"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list three
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane given a number of lists when a list edit button is clicked and the edit is cancelled should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list one
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-2"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list two
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-3"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list three
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane given a number of lists when a list edit button is clicked and the edit is completed should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list one
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-2"
      >
        <span
          class="title"
          data-testid="title"
        >
          new title for test list two
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-3"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list three
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane given a number of lists when a list edit button is clicked and the new list button is clicked should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list one
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-2"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list two
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-3"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list three
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-4"
      >
        <span
          class="title"
          data-testid="title"
        >
          <label
            for="list-title"
          >
            List Title
            <input
              class="ring-4 ring-opacity-30 ring-green-300"
              id="list-title"
              placeholder="List Title"
              type="text"
              value="New List"
            />
          </label>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane given a number of lists when a list edit button is clicked should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list one
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-2"
      >
        <span
          class="title"
          data-testid="title"
        >
          <label
            for="list-title"
          >
            List Title
            <input
              class="ring-4 ring-opacity-30 ring-green-300"
              id="list-title"
              placeholder="List Title"
              type="text"
              value="test list two"
            />
          </label>
        </span>
      </li>
      <li
        class="list"
        id="list-3"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list three
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane given a number of lists when the new list button is clicked should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list one
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-2"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list two
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-3"
      >
        <span
          class="title"
          data-testid="title"
        >
          test list three
        </span>
        <span
          class="action-btns"
        >
          <button
            class="p-1 border"
            type="button"
          >
            Edit
          </button>
          <button
            class="p-1 border"
            type="button"
          >
            Delete
          </button>
        </span>
      </li>
      <li
        class="list"
        id="list-4"
      >
        <span
          class="title"
          data-testid="title"
        >
          <label
            for="list-title"
          >
            List Title
            <input
              class="ring-4 ring-opacity-30 ring-green-300"
              id="list-title"
              placeholder="List Title"
              type="text"
              value="New List"
            />
          </label>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane initial render should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    />
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;

exports[`ListPane initial render when the new list button is clicked should render the expected html 1`] = `
<div>
  <div>
    <ol
      class="listPane"
    >
      <li
        class="list"
        id="list-1"
      >
        <span
          class="title"
          data-testid="title"
        >
          <label
            for="list-title"
          >
            List Title
            <input
              class="ring-4 ring-opacity-30 ring-green-300"
              id="list-title"
              placeholder="List Title"
              type="text"
              value="New List"
            />
          </label>
        </span>
      </li>
    </ol>
    <div
      class=""
    >
      <button
        aria-label="New List"
        class="p-1 bg-blue-100 border"
        type="button"
      >
        New List
      </button>
    </div>
  </div>
</div>
`;
